plugins/woocommerce/includes/admin/notes/class-wc-notes-run-db-update.php

```diff
@@ -202,22 +226,26 @@ private static function update_in_progress_notice( $note_id ) {
 		);
 
 		$note->save();
+
+		return $note->get_id();
 	}
 
 	/**
 	 * Update the existing note with $note_id with information that db upgrade is done.
 	 *
 	 * This is the last notice (3 out of 3 notices) displayed to the user.
 	 *
-	 * @param int $note_id Note id to update.
+	 * @param int|Note $note_id Note id to update.
 	 */
 	private static function update_done_notice( $note_id ) {
+		$note = ( $note_id instanceof Note ) ? $note_id : new Note( $note_id );
```

Comment:
I assume this is a small optimization to not re-initialize the note.  Could we update this to always use the `$note` instance if that's the case?  It seems like it's available in the other areas we use `update_done_notice()`.

Reply:
Right. For some reason I thought these were public methods and was trying to 'preserve' backwards compat, but they are private, so I've just reworked the logic a bit in 6b6a6f7609716df7b00d7b0baf653e50db68c22a. Hopefully it is now less confusing.